### ISSUE TITLE
Table bug: phantom borders on "noBorders" table

### ISSUE THREAD

**Issue by BirdInTheCity** [bpampuch/pdfmake#1088]
When setting a table to "noBorders" and giving it a background fill color, there is still a visible gap between cells that creates phantom borders.

Playground test example:
```
var dd = {
	content: [
		{
			style: 'tableExample',
			layout: 'noBorders',
			table: {
				body: [
					['Column 1', 'Column 2', 'Column 3'],
					['One value goes here', 'Another one here', 'OK?']
				]
			}
		}
	],
	styles: {
		
		tableExample: {
			margin: [0, 5, 0, 15],
			fillColor: "#2D4D81",
			color: "#ffffff"
		}
	}
}
```

Looks good in Chrome:
<img width="557" alt="screen shot 2017-07-29 at 12 08 42 pm" src="https://user-images.githubusercontent.com/814522/28746265-cdd5c894-7456-11e7-9cd7-1bfcd23386eb.png">

Phantom borders appear on iOS and in OSX Preview app:
<img width="556" alt="screen shot 2017-07-29 at 11 51 52 am" src="https://user-images.githubusercontent.com/814522/28746261-ba757a92-7456-11e7-8876-6a9d53c2a361.png">

**Comment by BirdInTheCity** [comment]
On a related note, I also tried matching the border color to the background color, and that also didn't work. It produced similar artifacts:

<img width="871" alt="screen shot 2017-07-29 at 12 37 44 pm" src="https://user-images.githubusercontent.com/814522/28746467-f32ddcc2-745a-11e7-941b-f306ed07ca57.png">

**Comment by liborm85** [comment]
I do not have a Mac so I have no way to test it...

**Comment by trueter** [comment]
Any chance we can bump this?

**Comment by BirdInTheCity** [comment]
In an effort to get this issue resolved, I posted a $15 bounty for it:
https://www.bountysource.com/issues/47733322-table-bug-phantom-borders-on-noborders-table

**Comment by BirdInTheCity** [comment]
@liborm85  I'm still trying to get this issue resolved, as it's making my pdf invoices look bad on iPhones/iPads/Macs. I'd like to dive deeper into the codebase and see if I can resolve it myself. Do you have any guesses as to where these draw calls are occurring?

With a little guidance, maybe I can fix?  0:)

**Comment by BirdInTheCity** [comment]
This is how the PDF invoice looks.  :(

![bad-looking](https://user-images.githubusercontent.com/814522/41719361-d45d294a-752d-11e8-9938-7b148c371495.png)

**Comment by BirdInTheCity** [comment]
It seems that it may just be a math error, where the cells are adding 1px or .5px padding.

I was able to "hide" the horizontal pixel by setting a 1px border, with the same border color as the background cell. However, the hack did not work for the vertical lines.
![vlines](https://user-images.githubusercontent.com/814522/41722712-4b1264ca-7537-11e8-8347-f7e364c6331b.png)


```
var dd = {

	content: [
		{
			style: 'tableExample',
			layout: 'noBorders',
			table: {
				body: [
					['Cell 1', 'Cell 2', 'Cell 3'],
					['Cell 4', 'Cell 5', 'Cell 6']
				]
			},
			layout: {
				hLineWidth: function (i, node) {
					return 1;
				},
				vLineWidth: function (i, node) {
					return (i === 0 || i === node.table.widths.length) ? 0 : 40;
				},
				hLineColor: function (i, node) {
					return '#2D4D81';
				},
				vLineColor: function (i, node) {
					return '#2D4D81';
				},
			}
		}
	],
	styles: {
		
		tableExample: {
			fillColor: "#2D4D81",
			color: "#ffffff"
		}
	}
}
```

**Comment by Radiergummi** [comment]
@BirdInTheCity your example seems to have a mistake - the layout property is duplicated. I'm stuck on a remotely similar issue - I'd like to reduce the `headerLineOnly` line width to 1px, but can't find a way to do so. Any idea how to go about that?

**Comment by BirdInTheCity** [comment]
@Radiergummi You're correct, there should be only one `layout` parameter. Though even with one, the error persists. BUT...

I found a hack to hide the vertical lines:
I set the `vLineWidth: -.01` See below:
```
var dd = {

	content: [
		{
			style: 'tableExample',
			table: {
				body: [
					['Cell 1', 'Cell 2', 'Cell 3'],
					['Cell 4', 'Cell 5', 'Cell 6']
				]
			},
			layout: {
				hLineWidth: function (i, node) {
					return 1;
				},
				vLineWidth: function (i, node) {
					return -.01;
				},
				hLineColor: function (i, node) {
					return '#2D4D81';
				},
				vLineColor: function (i, node) {
					return '#2D4D81';
				},
			}
		}
	],
	styles: {
		
		tableExample: {
			fillColor: "#2D4D81",
			color: "#ffffff"
		}
	}
}
```

 I'm guessing there's some math error in the table code. It's too complex for me to diagnose, but at least this masks it.

**Comment by liborm85** [comment]
Problem is identified.
Adobe Acrobat rendered correctly.
Google Chrome rendered correctly.
Mozilla PDF.js rendered **incorrectly**. This viewer contains render bug.

(source: https://github.com/bpampuch/pdfmake/pull/1631)

**Comment by BirdInTheCity** [comment]
@liborm85 Confirmed that the #1631 fix works! Please merge!  YAY!

**Comment by perbergland** [comment]
Still reproducable when viewed with system PDF renderer on macos

**Comment by perbergland** [comment]
Playground:

**Comment by modus-jose** [comment]
Any chance this will get merged, getting the same issue.
Any support I can provide to get it out? 
@BirdInTheCity or @liborm85

**Comment by modus-jose** [comment]
@perbergland , @BirdInTheCity I've tested this with 0.2.10 which uses jskit 0.14.0 
Which is the fix in the PR. It will help if you can confirm this is fixed on your side.

In my case the fix was to use 1px wide vertical lines with the same coloras the background in my layout.
